Source/Core/Common/x64Emitter.cpp

```diff
@@ -1572,6 +1579,69 @@ void XEmitter::CMP_or_TEST(int bits, const OpArg& a1, const OpArg& a2)
   }
 }
 
+void XEmitter::MOV_sum(int bits, X64Reg dest, const OpArg& a1, const OpArg& a2)
+{
+  // This stomps on flags, so ensure they aren't locked
+  _dbg_assert_(DYNA_REC, !flags_locked);
+
+  // If dest == a1 or dest == a2 we can simplify this
+  if (a1.IsSimpleReg() && a1.GetSimpleReg() == dest)
+  {
+    ADD(bits, R(dest), a2);
+    return;
+  }
+
+  if (a2.IsSimpleReg() && a2.GetSimpleReg() == dest)
+  {
+    ADD(bits, R(dest), a1);
+    return;
+  }
+
+  // Zero shortcuts
+  if (a1.IsZero())
+  {
+    MOV(bits, R(dest), a2);
+    return;
+  }
+  if (a2.IsZero())
+  {
+    MOV(bits, R(dest), a1);
+    return;
+  }
```

Comment:
I think we'd generate `mov reg, reg` which is effectively a NOP. It's probably more efficient than `add reg, 0` since it doesn't affect flags, but it might be useful to add that case in here.